keras/legacy/interfaces.py

```diff
@@ -74,3 +74,64 @@ def wrapper(*args, **kwargs):
 
         return func(*args, **kwargs)
     return wrapper
+
+
+def legacy_embedding_support(func):
+    """Function wrapper to convert the `Embedding` constructor from Keras 1 to 2.
+
+    # Arguments
+        func: `__init__` method of `Embedding`.
+
+    # Returns
+        A constructor conversion wrapper.
+    """
+    @six.wraps(func)
+    def wrapper(*args, **kwargs):
+        if len(args) > 3:
+            # The first entry in `args` is `self`.
+            raise TypeError('The `Embedding` layer can have at most '
+                            'two positional argument (`input_dim` and `output_dim`).')
+
+        # input_dim and output_dim
+        input_dim = kwargs.pop('input_dim', None)
+        if input_dim is None:
+            input_dim = args[1]
+
+        output_dim = kwargs.pop('output_dim', None)
+        if output_dim is None:
+            output_dim = args[2]
+
+        args = (args[0], input_dim, output_dim)
+
+        converted = []
+
+        # Remaining kwargs.
+        conversions = [
+            ('init', 'embeddings_initializer'),
+            ('W_regularizer', 'embeddings_regularizer'),
+            ('W_constraint', 'embeddings_constraint'),
```

Comment:
Additionally, the old `Embedding` supported a `dropout` argument that we no longer support. Since calls using the old API should still work, the dropout argument needs to be handled. Just remove it from kwargs if present, and raise a warning saying that the argument no longer exists that people should use instead a `keras.layers.SpatialDropout1D` layer right after the `Embedding` layer to get the same behavior.